genson/src/main/java/com/owlike/genson/GensonBuilder.java

```diff
@@ -60,6 +60,7 @@
   private VisibilityFilter methodFilter = VisibilityFilter.PACKAGE_PUBLIC;
   private VisibilityFilter constructorFilter = VisibilityFilter.PACKAGE_PUBLIC;
 
+  private ClassLoader classLoader = null;
```

Comment:
Why not initialize it to `GensonBuilder.getClassLoader` and then just do new `ScalaBeanPropertyFactory(builder.getClassLoader())`? An additional not null check could be added to withClassLoader method.